app/src/main/java/org/listenbrainz/android/presentation/features/dashboard/DashboardActivity.kt

```diff
@@ -53,21 +57,37 @@ class DashboardActivity : ComponentActivity() {
             ListenBrainzTheme()
             {
                 var isGrantedPerms by remember {
-                    mutableStateOf(false)
+                    mutableStateOf(preferencePermsGranted)
```

Comment:
What's the reason for duplicating this here? Can't we just use one variable here?

Reply:
The dialog appears for some brief moment before every launch. to solve the issue i did this. Plus this can bypass the use of checkSelfPermission by introducing a simple variable access.

Reply:
Alright, we do need a mutable state here.

Reply:
Im doing a solution waiktt

Reply:
Haha! Can submit another PR on top of this